docs/en/snapshotcloneserver_en.md

```diff
@@ -0,0 +1,165 @@
+
+The following steps are for creating a snapshot:
+
+1. Generate snapshot records and persist them to Etcd. Several judgments are required for this step: For the same volume, there can only be one snapshot being taken at the same time, so it is necessary to judge whether the volume has a snapshot request being processed. Theoretically, unlimited snapshots are supported by the system, and in the implementation we put a limit to the depth of snapshot, which can be modified in the configuration file. For this reason, we also need to determine whether the number of snapshot exceeds the limit. If every thing works well, the metadata of the original volume will be read from MDS and a snapshot record will be generated and persisted to Etcd. The status of the snapshot generated in this step is 'pending'.
+
+2. Create temporary snapshot on CurveFS, return seqNum of the snapshot created, then update the seqNum of the snapshot record.
+4. Create snapshot mapping table (see 1.4.1), and save it to object storage S3.
+6. Dump snapshot data from CurveFS to object storage S3. When storing, the snapshot system first reads the snapshot data from CurveFS, then uploads the data to the S3.
+8. Delete the temporary snapshot.
+10. Update snapshot status to 'done'.
+
+### 1.4 Organization of Snapshot Data
+
+Each snapshot has a corresponding record in the snapshot system which persisted in Etcd and use a uuid as unique identifier.
+
+#### 1.4.1 Organization of snapshot data on S3
+
+Snapshot data is stored in S3 in chunks, and each chunk corresponds to an object on S3. The object that stores data is called a data object, and each snapshot also has an meta object that records the metadata. The meta object records the file information of the snapshot and the mapping table between the snapshot chunk and the object. The meta object consists of 2 parts:
+
+1. Snapshot file information, including original data volume name, volume size and chunk size, etc.
+2. Snapshot chunk mapping table, recording the list of snapshot data objects.
+
+Figure 2 is a sample of the sanpshot data.
+
+<p align="center">
+    <img src="../images/snap-s3-format.png"><br>
+    <font size=3> Figure 3: Snapshot data format</font>
+</p>
+
+### 1.5 Interface of Curve Snapshot System
+
+Please refer to the [document](../cn/snapshotcloneserver_interface.md) of snapshot and clone.
+
+## 2、Clone
+
+### 2.1 Intro to Curve Clone Module
+
+In section 1 we introduced the snapshot system, which is for creating snapshots of  files of CurveFS and incrementally dump the data to S3. But for a complete snapshot system, supporting only the snapshot creation is not enough, file recovery and cloning is the reason why take snapshot. In this section we introduce Curve cloning systerm(recovering can be considered as cloning to some extend).
```

Comment:
recovering can be considered as cloning to some extend
recovering can be considered as cloning to some **extent**

Reply:
> recovering can be considered as cloning to some extend
> recovering can be considered as cloning to some **extent**

done